iceoryx_examples/user_header/publisher_cxx_api.cpp

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) 2021 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+//! [iceoryx includes]
+#include "user_header_and_payload_types.hpp"
+
+#include "iceoryx_posh/popo/publisher.hpp"
+#include "iceoryx_posh/runtime/posh_runtime.hpp"
+#include "iceoryx_utils/posix_wrapper/signal_handler.hpp"
+//! [iceoryx includes]
+
+#include <atomic>
+#include <iostream>
+
+//! [signal handling]
+std::atomic<bool> killswitch{false};
```

Comment:
Nitpick `std::atomic_bool`

Reply:
It's just a personal preference to use the template variant and would like to keep it if you don't have hard feelings

Reply:
Well, for me our example follow the style we suggest the users to use. My personal preference would be the usage of `std::atomic_bool`, but no hard feelings here ;)